_lib/core/shop.php

```diff
@@ -246,15 +284,15 @@ public function get_basket()
         $this->vat = $this->subtotal * $this->vat_rate;
     }
 
-    public function add_to_basket($product, $quantity = 1, $buy_now = false, $variation_id = null, $extras_arr = '')
+    public function add_to_basket($product, int $quantity = 1, bool $buy_now = false, $variation_id = null, $extras_arr = '')
```

Comment:
@adamjimenez What type is product, variation_id and extras_arr?

Reply:
product = int
variation_id = int
extras_arr = array